server/src/snapshot_middleware/rbxm.rs

```diff
@@ -0,0 +1,96 @@
+use std::{
+    borrow::Cow,
+    collections::HashMap,
+};
+
+use rbx_dom_weak::{RbxTree, RbxInstanceProperties, RbxId};
+
+use crate::{
+    imfs::new::{Imfs, ImfsFetcher, ImfsEntry},
+    snapshot::InstanceSnapshot,
+};
+
+use super::{
+    middleware::{SnapshotMiddleware, SnapshotInstanceResult, SnapshotFileResult},
+};
+
+pub struct SnapshotRbxm;
+
+impl SnapshotMiddleware for SnapshotRbxm {
+    fn from_imfs<F: ImfsFetcher>(
+        imfs: &mut Imfs<F>,
+        entry: &ImfsEntry,
+    ) -> SnapshotInstanceResult<'static> {
+        if entry.is_directory() {
+            return Ok(None);
+        }
+
+        let file_name = entry.path()
+            .file_name().unwrap().to_string_lossy();
+
+        if !file_name.ends_with(".rbxm") {
+            return  Ok(None);
```

Comment:
Double space. `cargo fmt` :eyes:

Reply:
Yeaah, seems like it'd be a good idea to `cargo fmt` the whole codebase consistently after this PR is merged.

Reply:
That'll fix the double space too, I s'pose.